var/spack/repos/builtin/packages/py-htgettoken/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyHtgettoken(PythonPackage):
+    """htgettoken gets OIDC authentication tokens for High Throughput Computing
+    via a Hashicorp vault server."""
+
+    homepage = "https://github.com/fermitools/htgettoken"
+
+    # htgettoken is not available on PyPi
+    url = "https://github.com/fermitools/htgettoken/archive/refs/tags/v1.16.tar.gz"
+    list_url = "https://github.com/fermitools/htgettoken/releases/"
+    git = "https://github.com/fermitools/htgettoken.git"
+
+    maintainers = ["wdconinc"]
+
+    version("1.16-33-g3788bb4", commit="3788bb4733e5e8f856cee51566df9a36cbfe097d")
+    version("1.16-20-g8b72f48", commit="8b72f4800ef99923dac99dbe0756a26266a27886")
```

Comment:
Are these official version numbers? Might be better to use something like `1.16-2022-02-06` that sorts better. Could also just only include master only and wait for a new release

Reply:
1.16 is an official version number, but doesn't have setuptools. `8b72f4800ef99923dac99dbe0756a26266a27886` is when setuptools support was added. `3788bb4733e5e8f856cee51566df9a36cbfe097d` is the 'natural' commit after a bunch of changes. These are reproducible lexically sortable version numbers from `git describe`. As soon as 1.17 is out, we'd just refer to that and forget about the 'special' `git describe` versions.

Reply:
Added documentation to package.py.